workers/proto/kitchen_sink/kitchen_sink.proto

```diff
@@ -0,0 +1,328 @@
+  google.protobuf.Duration schedule_to_close_timeout = 6;
+  // Limits time an activity task can stay in a task queue before a worker picks it up. This
+  // timeout is always non retryable as all a retry would achieve is to put it back into the same
+  // queue. Defaults to schedule_to_close_timeout or workflow execution timeout if not specified.
+  google.protobuf.Duration schedule_to_start_timeout = 7;
+  // Maximum time an activity is allowed to execute after a pick up by a worker. This timeout is
+  // always retryable. Either this or schedule_to_close_timeout must be specified.
+  google.protobuf.Duration start_to_close_timeout = 8;
+  // Maximum time allowed between successful worker heartbeats.
+  google.protobuf.Duration heartbeat_timeout = 9;
+  // Activities are provided by a default retry policy controlled through the service dynamic
+  // configuration. Retries are happening up to schedule_to_close_timeout. To disable retries set
+  // retry_policy.maximum_attempts to 1.
+  temporal.api.common.v1.RetryPolicy retry_policy = 10;
+  // Whether or not this activity will be a local activity
+  oneof locality {
+    google.protobuf.Empty is_local = 11;
+    RemoteActivityOptions remote = 12;
+  }
+  // If set, when to cancel the activity
+  CancelCondition cancel_condition = 13;
+}
+
+message ExecuteChildWorkflowAction {
+  string namespace = 2;
+  string workflow_id = 3;
+  string workflow_type = 4;
+  string task_queue = 5;
+  repeated temporal.api.common.v1.Payload input = 6;
+  // Total workflow execution timeout including retries and continue as new.
+  google.protobuf.Duration workflow_execution_timeout = 7;
+  // Timeout of a single workflow run.
+  google.protobuf.Duration workflow_run_timeout = 8;
+  // Timeout of a single workflow task.
+  google.protobuf.Duration workflow_task_timeout = 9;
+  // Default: PARENT_CLOSE_POLICY_TERMINATE.
+  ParentClosePolicy parent_close_policy = 10;
+  // string control = 11; (unused from StartChildWorkflowExecutionCommandAttributes)
+  // Default: WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE.
+  temporal.api.enums.v1.WorkflowIdReusePolicy workflow_id_reuse_policy = 12;
+  temporal.api.common.v1.RetryPolicy retry_policy = 13;
+  string cron_schedule = 14;
+  // Header fields
+  map<string, temporal.api.common.v1.Payload> headers = 15;
+  // Memo fields
+  map<string, temporal.api.common.v1.Payload> memo = 16;
+  // Search attributes
+  map<string, temporal.api.common.v1.Payload> search_attributes = 17;
+  // Defines behaviour of the underlying workflow when child workflow cancellation has been requested.
+  temporal.omes.kitchen_sink.child_workflow.ChildWorkflowCancellationType cancellation_type = 18;
+  // Whether this child should run on a worker with a compatible build id or not.
+  VersioningIntent versioning_intent = 19;
+}
+
+message AwaitSignalAction {
+  string signal_name = 1;
+}
+
+message AwaitUpdateAction {
+  string update_name = 1;
+}
```

Comment:
I'm thinking actually I could replace both of these with `AwaitWorkflowStateCondition` and a key/value pair.